relayer/src/foreign_client.rs

```diff
@@ -230,20 +230,13 @@ pub fn build_update_client_and_send(
     )?;
 
     let mut events = dst_chain.send_msgs(new_msgs)?;
-    assert!(!events.is_empty());
+    assert_eq!(events.len(), 1);
     Ok(events.pop().unwrap())
 }
 
 /// Tests the integration of crates `relayer` plus `relayer-cli` against crate `ibc`. These tests
 /// exercise various client methods (create, update, ForeignClient::new) using locally-running
 /// instances of chains built using `MockChain`.
-///
-/// ## Why are all these tests ignored?
-/// We ignore these tests as of #451, because the mock chain is not yet capable of producing
-/// transaction responses of correct types (the types should be similar to `tx_commit::Response`).
-/// Another problem is that `build_create_client_and_send` contains a Cosmos-specific method
-/// for parsing transaction response. Once this parsing stage is general enough for the Mock chain,
-/// these tests should require minimal changes to pass.
```

Comment:
I think it would be a good idea to update the tests here to match on the type of the event that `build_update_client_and_send` returns (optionally extract and assert on the client identifier from the event).

Reply:
Good idea! Done in e0db63c8e11d11a804910d3f9a813547beadb31a.